test/unit/html/HtmlConverterTest.php

```diff
@@ -238,6 +238,55 @@ public function testInsertSnippetAndHreflangTagsWithFallback()
         $this->assertEquals($expected_html, $translated_html);
     }
 
+    public function testInsertHtmlLangAttribute()
+    {
+        $html_cases = array(
+            array(
+                'general case - insert lang attribute',
+                '<html><head></head><body><a>hello</a></body></html>',
+                '<html lang="en"><head></head><body><a>hello</a></body></html>',
+            ),
+            array(
+                'html with other attribute - insert lang attribute',
+                '<html test="lang"><head></head><body><a>hello</a></body></html>',
+                '<html lang="en" test="lang"><head></head><body><a>hello</a></body></html>',
+            ),
+            array(
+                'lang attribute exists - keep existing lang',
+                '<html lang="ja"><head></head><body><a>hello</a></body></html>',
+                '<html lang="ja"><head></head><body><a>hello</a></body></html>',
+            ),
+            array(
+                'lang attribute exists with single quotes - keep existing lang',
+                "<html lang='ja'><head></head><body><a>hello</a></body></html>",
+                "<html lang='ja'><head></head><body><a>hello</a></body></html>",
+            ),
+            array(
+                'lang attribute exists without quotes - keep existing lang',
+                "<html lang=ja><head></head><body><a>hello</a></body></html>",
+                "<html lang=ja><head></head><body><a>hello</a></body></html>",
+            ),
+            array(
+                'lang code has dash - keep existing lang',
+                '<html lang="zh-CHS"><head></head><body><a>hello</a></body></html>',
+                '<html lang="zh-CHS"><head></head><body><a>hello</a></body></html>',
+            )
+        );
+        foreach ($html_cases as $case) {
+            list($message, $original_html, $expected_html) = $case;
+            $settings = array(
+                'supported_langs' => array('en', 'vi'),
+                'lang_param_name' => 'wovn',
+                'insert_hreflangs' => false
+            );
+            list($store, $headers) = StoreAndHeadersFactory::fromFixture('default', $settings);
+            $converter = new HtmlConverter('UTF-8', $store->settings['project_token'], $store, $headers);
+            $translated_html = $converter->insertHtmlLangAttribute($original_html, 'en');
```

Comment:
When you change these test cases to call the public method instead, I would recommend changing the assertion to `translated_html.contains('<html lang="...">`) instead.

Reply:
I fixed by `ReflectionMethod` :stuck_out_tongue_closed_eyes:

Reply:
That's not really good practice either.

Reply:
I changed tests.
I think these tests are much better than before.

Reply:
great. I think you can merge it now.